samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt

```diff
@@ -104,16 +104,17 @@ class DrumThumperActivity : AppCompatActivity(), TriggerPad.DrumPadTriggerListen
 
         mAudioMgr = getSystemService(Context.AUDIO_SERVICE) as AudioManager
 
-        if (mUseDeviceChangeFallback) {
-            mAudioMgr!!.registerAudioDeviceCallback(mDeviceListener, null)
-        }
-
-        mDrumPlayer.setupAudioStream()
+        mDrumPlayer.allocSampleData()
         mDrumPlayer.loadWavAssets(getAssets())
     }
 
     override fun onStart() {
         super.onStart()
+        if (mUseDeviceChangeFallback) {
```

Comment:
The start and stop are still not symmetric.  Is ABC-BCA, should probably be ABC-CBA
Can there be a problem if the device callback occurs before the audio is setup?
Or vice versa.